### PR TITLE
[ADD] l10n_cl_hr_expense: add module to fix hr_expense issue with chi…

### PR DESCRIPTION
**Pull request by Danisan**
Description of the issue/feature this PR addresses:
Incompatibility between current chilean localization and expenses modules, due to need of use documents in purchase journals.

Current behavior before PR:
When installed l10n_cl and hr_expense, when trying to create expenses entries in purchase journal from expense app, you will have a UserError saying that you should define documents.

Desired behavior after PR is merged:
It force to use miscellaneus journal instead of purchase journal for a chilean localized company, changing the domain and the default journal selection.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_cl_hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        """ The journal is determining the company of the accounting entries generated from expense. We need to
+        force journal company and expense sheet company to be the same. """
+        journal_domain = self._get_journal_domain()
+        journal = self.env['account.journal'].search(journal_domain, limit=1)
+        return journal.id
+
+    journal_id = fields.Many2one(
+        'account.journal', string='Expense Journal',
+        states={'done': [('readonly', True)], 'post': [('readonly', True)]}, check_company=True,
+        domain=_get_journal_domain, default=_default_journal_id, help="The journal used when the expense is done.")
```
Comment (william-andre) [inline addons/l10n_cl_hr_expense/models/hr_expense.py:27]: Define `_get_journal_domain` in `hr_expense` instead and don't change the field declaration here

File: addons/l10n_cl_hr_expense/models/hr_expense.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    @api.model
+    def _get_journal_domain(self):
+        default_company_id = self.default_get(['company_id'])['company_id']
+        if self.env['res.company'].browse(default_company_id).country_id.code == 'CL':
+            return [('type', '=', 'general'), ('company_id', '=', default_company_id)]
+        return [('type', '=', 'purchase'), ('company_id', '=', default_company_id)]
```
Comment (william-andre) [inline addons/l10n_cl_hr_expense/models/hr_expense.py:14]: Return `super()._get_journal_domain()` instead

File: addons/l10n_cl_hr_expense/models/hr_expense.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    @api.model
+    def _get_journal_domain(self):
+        default_company_id = self.default_get(['company_id'])['company_id']
+        if self.env['res.company'].browse(default_company_id).country_id.code == 'CL':
+            return [('type', '=', 'general'), ('company_id', '=', default_company_id)]
```
Comment (william-andre) [inline addons/l10n_cl_hr_expense/models/hr_expense.py:13]: This domain won't be the right one if you change the company in the form view (multi-company settings)

File: addons/l10n_cl_hr_expense/models/hr_expense.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        return [('type', '=', 'purchase'), ('company_id', '=', default_company_id)]
+
+    @api.model
+    def _default_journal_id(self):
+        """ The journal is determining the company of the accounting entries generated from expense. We need to
+        force journal company and expense sheet company to be the same. """
+        journal_domain = self._get_journal_domain()
+        journal = self.env['account.journal'].search(journal_domain, limit=1)
+        return journal.id
```
Comment (william-andre) [inline addons/l10n_cl_hr_expense/models/hr_expense.py:22]: This should be in `hr_expense` too

**Review by william-andre — CHANGES_REQUESTED** [review]
Basically the only thing that should be in this module is the extension of `_get_journal_domain` for `l10n_cl`

Also need to find a solution do have a dynamic domain depending on the right company; which might not be the default one.

File: addons/l10n_cl_hr_expense/models/hr_expense.py (hunk 3 above)
Comment (Danisan) [reply addons/l10n_cl_hr_expense/models/hr_expense.py:13]: Thanks @william-andre, made the suggested changes in 606020f8c531b2c7abfe6c19d3fc97848a5bbd45

**Comment by vbe-odoo** [comment]
@william-andre hi, can you check this PR, @Danisan already did the changes suggested.
Thanks!
cc' @jco-odoo @suv-bmya

**Review by william-andre — CHANGES_REQUESTED** [review]
CI/template is red too

**Comment by Danisan** [comment]
This is replaced by #76915 with a different perspective. So could be closed if that PR is accepted

**Pull request closed without merging**